src/vcpkg/base/downloads.cpp

```diff
@@ -1012,11 +1016,6 @@ namespace vcpkg
                 }
             }
         }
-        // Asset cache is not configured and x-block-origin enabled
```

Comment:
The previous comment is incorrect: the problem is not every code path on line 988 `if (!m_config.m_block_origin)` returns or exits. We probably need to add a `if (m_config.m_block_origin)` conditional here or refactor both sections.